src/main/java/org/ethereum/beacon/discovery/schema/NodeRecord.java

```diff
@@ -104,9 +104,16 @@ public static NodeRecord fromRawFields(
     return nodeRecord;
   }
 
-  @SuppressWarnings({"unchecked", "DefaultCharset"})
   public String asBase64() {
-    return new String(Base64.getUrlEncoder().encode(serialize().toArray()));
+    String result = Base64.getUrlEncoder().encodeToString(serialize().toArray());
+    while (result.endsWith("=")) {
```

Comment:
Is it a bad idea to just search for the first index of "=" and truncate from there?

Reply:
No that makes sense. `=` will only ever appear as a padding char at the so that would be faster and simpler.